### ISSUE TITLE
Wrong alignment of label text in macOS

### ISSUE THREAD

**Issue by SamSchott** [beeware/toga#1029]
**Describe the bug**

The text in a `toga.Label` is offset by roughly 3 pixels in Cocoa when no background color is set, resulting in a misalignment with other widgets. This issue was introduced in PR #1002 and comes from the behaviour of `NSTextField` when setting `native.drawsBackground = True`: when a background is drawn, even if `backgroundColor = None`, the text is offset to right so that the left edge of the background color is not flush with the text itself. This makes sense when the background is actually colored but results in an annoying apparent misalignment with other UI elements if `backgroundColor = None`.

I can see two possible ways to fix this:

1. Override `set_background` in `toga_cocoa.Label` to set `native.drawsBackground = False` by default, i.e., when `background_color = None`.

2. Revert this PR and implement `set_background` for each widget individually to avoid possible issues with other existing or future new widgets.

Both fixes have the small disadvantage that the Label is drawn with transparent background despite `background_color != TRANSPARENT`, as was the case for the previous implementation as well.

**To Reproduce**
Create a regular `toga.Label` on macOS and compare its left edge to other widgets in the same column.

<details><summary>Example code</summary>
<p>

```python
import toga
from toga.constants import COLUMN, TRANSPARENT, GREEN
from toga.style import Pack


class LabelTestApp(toga.App):

    def startup(self):
        self.main_window = toga.MainWindow(title=self.name)

        self.label0 = toga.Label('Green background', style=Pack(background_color=GREEN))
        self.label1 = toga.Label('No background', style=Pack(background_color=TRANSPARENT))
        self.label2 = toga.Label('Default background')

        box = toga.Box(
            children=[
                self.label0,
                self.label1,
                self.label2,
            ],
            style=Pack(
                flex=1,
                direction=COLUMN,
                padding=10
            )
        )

        self.main_window.content = box
        self.main_window.show()


def main():
    return LabelTestApp('Labels', 'org.beeware.labels')


if __name__ == '__main__':
    app = main()
    app.main_loop()

```
</p>
</details>

**Screenshots**
<img width="344" alt="Screenshot 2020-08-19 at 16 30 34" src="https://user-images.githubusercontent.com/20974290/90649480-131f7d00-e23b-11ea-8d22-aed6ea8ee1a9.png">

**Environment:**
 - Operating System: macOS Catalina and higher
 - Python version: 3.8
 - Software versions:
   - Toga: 0.3.0.dev23

**Comment by alexaatm** [comment]
Was able to reproduce the issue. Option 3 seems the cleanest in the long run, but the most invasive. Is there a preferred way on the maintainers' side?

**Comment by freakboy3742** [comment]
I agree that option 3 seems like the right approach - and would have the added benefit of being consistent with the direction that #767 is leading.

**Comment by alexaatm** [comment]
Gonna work on it during Pycon Sprint 2025:)
